pkg/mount/mountutil.go

```diff
@@ -0,0 +1,151 @@
+	mountAttr, err := m.getDeviceMountAttr(m.getPartitionName)
+	if err != nil {
+		return "", err
+	}
+	devPath, err := getDiskDevPath(mountAttr.DevPath)
+	if err != nil {
+		return "", err
+	}
+	_, err = filepath.EvalSymlinks(devPath)
+	if err != nil {
+		return "", err
+	}
+	return devPath, err
+}
+
+// getDeviceMountAttr read mounts file and returns device mount attributes, which includes partition name,
+// mountpoint and filesystem
+func (m MountUtil) getDeviceMountAttr(fn getMountData) (DeviceMountAttr, error) {
+	mountAttr := DeviceMountAttr{}
+	// Read file from filepath and get which partition is mounted on given mount point
+	file, err := os.Open(m.filePath)
+	if err != nil {
+		return mountAttr, err
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		return mountAttr, err
+	}
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		/*
+			read each line of given file in below format -
+			/dev/sda4 / ext4 rw,relatime,errors=remount-ro,data=ordered 0 0
+			/dev/sda4 /var/lib/docker/aufs ext4 rw,relatime,errors=remount-ro,data=ordered 0 0
+		*/
+
+		// we are interested only in lines that start with /dev
+		if !strings.HasPrefix(line, "/dev") {
+			continue
+		}
+		if mountAttr, ok := fn(line); ok {
+			return mountAttr, nil
+		}
+	}
+	return mountAttr, fmt.Errorf("could not get device mount attributes, Path/MountPoint not present in mounts file")
+}
+
+//	getDiskSysPath takes disk/partition name as input (sda, sda1, sdb, sdb2 ...) and
+//	returns syspath of that disk from which we can generate ndm given uuid of that disk.
+func getDiskDevPath(partition string) (string, error) {
+	// dev path be like /dev/sda4 we need to remove /dev/ from this string to get sys block path.
+	var diskName string
+	softlink := "/sys/class/block/" + partition
+	link, err := filepath.EvalSymlinks(softlink)
+	if err != nil {
+		return "", err
+	}
+	/*
+		link looks - /sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda4
+		parent disk is present after block then partition of that disk
+	*/
+	parts := strings.Split(link, "/")
+	for i, part := range parts {
+		if part == "block" {
+			diskName = parts[i+1]
+			break
+		}
+	}
+	return "/dev/" + diskName, nil
+}
+
+// getPartitionName gets the partition name from the mountpoint. Each line of a mounts file
+// is passed to the function, which is parsed and partition name is obtained
+func (m *MountUtil) getPartitionName(mountLine string) (DeviceMountAttr, bool) {
+	mountAttr := DeviceMountAttr{}
+	isValid := false
+	if len(mountLine) == 0 {
+		return mountAttr, isValid
+	}
+	if parts := strings.Split(mountLine, " "); parts[1] == m.mountPoint {
+		mountAttr.DevPath = strings.Replace(parts[0], "/dev/", "", 1)
+		isValid = true
+	}
+	return mountAttr, isValid
+}
+
+// getMountName gets the mountpoint, filesystem etc from the partition name. Each line of a mounts
+// file is passed to the function, which is parsed to get the information
```

Comment:
Provide an example of the mountline. Will there be a cases, where devices can be mounted with mount options and are those handled by this function?

Reply:
@kmova  when devices are mounted with mount options, those options will be ignored currently.

Reply:
Provided sample `mountline`